### PR TITLE
treewide: change `${pname}` to string literal

### PR DESCRIPTION
**Pull request by pbsds**
## Description of changes

please squash merge this, i spam-commit for easier handling of merge conflicts.

done with

```shell
#!/usr/bin/env nix-shell
#!nix-shell --pure -i bash -p ripgrep sd jq git lix

export NIXPKGS_ALLOW_UNFREE=1
export NIXPKGS_ALLOW_INSECURE=1
export NIXPKGS_ALLOW_BROKEN=1

git restore .

test -s packages.json || ( set -x;
  time nix-env --extra-experimental-features no-url-literals --option system x86_64-linux -f ./. -qaP --json --meta --show-trace --no-allow-import-from-derivation > packages.json
)

jq <packages.json 'to_entries[] | select(.value.meta.position==null|not) | "\(.key)\t\(.value.meta.position)"' -r |
    sed -e "s#\t$(realpath .)/#\t#" |
    sed -e 's#:\([0-9]*\)$#\t\1#' |
    grep --fixed-strings "$( rg -F '${pname}' pkgs/ -tnix -l | sort | head -n 2500 )" |
    grep . |
    grep -iv haskell |
    grep -iv /top-level/ |
    grep -iv chicken |
    grep pkgs/by-name/ |
    grep -iv build |
    grep -E '/(package|default)\.nix' |
    while read attrpath fname col; do
        echo "$attrpath"
        data="$(nix eval --impure  --expr 'with import ./. {}; { inherit ('"$attrpath"') pname drvPath passthru meta; drvPath2='"$attrpath"'.src.drvPath; }' --json)"
        test -n "$data" || continue
        pname="$(jq <<<"$data" .pname -r)"
        test -n "$pname" || continue
        (set -x
            sd -F '${pname}' "$pname" "$fname"
            sd -F ' = pname;' " = \"$pname\";" "$fname"
        )
        data2="$(nix eval --impure  --expr 'with import ./. {}; { inherit ('"$attrpath"') pname drvPath passthru meta; drvPath2='"$attrpath"'.src.drvPath; }' --json)"
        if ! test "$data" = "$data2"; then
            (set -x; git restore "$fname")
        fi
    done
```

in `pkgs/by-name` this time around, and i selected the ones that made sense using `GIT_DIFF_OPTS=-u2000 git add --patch`.

I avoided various patterns like `name = "${pname}-${version}"` because that would hamper a different kind of treewide, and other patterns where `${pname}` is being used in passthru.tests or fod storepath names, and in assertions that makes sense to copy to dervations.

should be 0 rebuilds

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/br/braa/package.nix [hunk 1]
```diff
@@ -9,7 +9,7 @@ stdenv.mkDerivation rec {
   version = "0.82";
 
   src = fetchzip {
-    url = "http://s-tech.elsat.net.pl/${pname}/${pname}-${version}.tar.gz";
+    url = "http://s-tech.elsat.net.pl/braa/braa-${version}.tar.gz";
```
Comment (philiptaron) [inline pkgs/by-name/br/braa/package.nix:12]: lol

File: pkgs/by-name/vv/vvvvvv/package.nix [hunk 2]
```diff
@@ -59,7 +59,7 @@ stdenv.mkDerivation rec {
       name = "VVVVVV";
       desktopName = "VVVVVV";
       comment = meta.description;
-      exec = pname;
+      exec = "vvvvvv";
       icon = "VVVVVV";
```
Comment (philiptaron) [inline pkgs/by-name/vv/vvvvvv/package.nix:63]: golly

**Review by philiptaron — APPROVED** [review]
I intend to merge once OfBorg comes back clean.

**Comment by sternenseemann** [comment]
The motivation you have given previously for this sort of change, https://github.com/NixOS/nixpkgs/pull/387725#issuecomment-2924073216 (which also doesn't make sense in all possible cases), doesn't apply for `rec` set `pname` etc. at all. Why do this?

**Comment by pbsds** [comment]
Most derivations are one drive-by contribution or review nitpick away from migrating to `finalAttrs`, as such I treat `let-in` and `rec` as if they are overrideable.

**Comment by pbsds** [comment]
these threads champion a _correctness_ argument:

* https://github.com/NixOS/nixpkgs/issues/277994
* https://github.com/NixOS/nixpkgs/issues/310373

**Comment by pbsds** [comment]
> Why do this?

One of my motivations for https://github.com/NixOS/nixpkgs/issues/346453 is to show people that most of the common nitpicks are trivially fixed with scripted treewides, and should as such not cause unnecessary review cycles.

**Pull request merged**